beacon-chain/blockchain/setup_forchoice.go

```diff
@@ -0,0 +1,84 @@
+package blockchain
+
+import (
+	"bytes"
+
+	"github.com/pkg/errors"
+	forkchoicetypes "github.com/prysmaticlabs/prysm/v5/beacon-chain/forkchoice/types"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/features"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+)
+
+func (s *Service) setupForkchoice(st state.BeaconState) error {
+	if err := s.setupForkchoiceCheckpoints(); err != nil {
+		return errors.Wrap(err, "could not set up forkchoice checkpoints")
+	}
+	if err := s.setupForkchoiceRoot(st); err != nil {
+		return errors.Wrap(err, "could not set up forkchoice root")
+	}
+	if err := s.initializeHeadFromDB(s.ctx, st); err != nil {
+		return errors.Wrap(err, "could not initialize head from db")
+	}
```

Comment:
This isn't a blocker for me, just a review comment. We could have retrieved the block from the DB outside these calls with `ensureRootNotZeros` and then passed it to `setupForkchoiceRoot` and `initializeHeadFromDB`. Right now, we're calling `ensureRootNotZeros` and `BeaconDB.Block` twice.  

That said, I understand you probably don’t want to change the logic too much